packages/react/src/primitives/utils/useStableId.ts

```diff
@@ -0,0 +1,31 @@
+// Adapted from https://github.com/radix-ui/primitives/blob/main/packages/react/id/src/id.tsx#L8
+import * as React from 'react';
+import { useLayoutEffect } from './useLayoutEffect';
+
+// Prefixed autogenerated id created by useStableId
+export const AUTO_GENERATED_ID_PREFIX = 'amplify-id';
+
+export type UseStableId = (id?: string) => string;
+
+// Create a local version of React.useId which will reference React.useId for React 18
+// and fallback to noop for React 17 and below
+const useReactId = (React as any).useId || (() => undefined);
```

Comment:
New React `useId` function always return a `string` > https://github.com/facebook/react/blob/4729ff6d1f191902897927ff4ecd3d1f390177fa/packages/react-reconciler/src/ReactFiberHooks.new.js#L2035

Reply:
Not sure if you're trying to suggest a change here. The reason the noop function returns `undefined` is to make sure the `id` is `undefined` on SSR and first render to prevent hydration mismatch issues.

Reply:
aaaah, got it, thanks for the clarification